classes/submission/Collector.php

```diff
@@ -510,8 +512,10 @@ public function getQueryBuilder(): Builder
         }
 
         if (isset($this->categoryIds)) {
-            $q->join('publication_categories as pc', 's.current_publication_id', '=', 'pc.publication_id')
-                ->whereIn('pc.category_id', $this->categoryIds);
+            $publicationIds = PublicationCategory::whereIn('category_id', $this->categoryIds)
```

Comment:
For code external to the `PublicationCategory` class, better to introduce local scope functions and call them from outside.